### PR TITLE
RBSM header matching

### PR DESCRIPTION
**Pull request by VivaanBahl**
Was tested to make sure that it initializes properly, still need to test
to make sure it can retrieve and validate properly

### CODE REVIEW

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java [hunk 1]
```diff
@@ -1,45 +1,79 @@
...
+	 */
+	public static boolean initializeHeaders() {
+
+		headers = new JsonObject();
+
+		try {
+			Scanner fileIn = new Scanner(new File(RobobuggyConfigFile.RBSM_HEADER_FILE_LOCATION), "UTF-8");
+			while (fileIn.hasNextLine()) {
+				String line = fileIn.nextLine();
+				if (!line.equals("")) {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2165077905/issues/source?bid=2938665&fileBranchId=3160458#l35)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
     //Library info
     public static final String ALICE_LIBRARY_VERSION = "1.0.0";
+	public static final String RBSM_HEADER_FILE_LOCATION = "../../../arduino_src/lib_avr/rbserialmessages/rbsm_headers.txt";
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid excessively long variable names like RBSM_HEADER_FILE_LOCATION](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2165077860/issues/source?bid=2938665&fileBranchId=3160458#l12)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java [hunk 3]
```diff
@@ -239,7 +239,7 @@ public void interrupt() {
         /**
          * Instantiates a new LogWriterThread by clearing the message queue
          */
-        public LogWriterThread() {
+        LogWriterThread() {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2166439835/issues/source?bid=2938665&fileBranchId=3160458#l242)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid excessively long variable names like RBSM_HEADER_FILE_LOCATION](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2166439796/issues/source?bid=2938665&fileBranchId=3160458#l12)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2166439841/issues/source?bid=2938665&fileBranchId=3160458#l35)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java [hunk 4]
```diff
@@ -1,45 +1,79 @@
...
-	public static final byte BATTERY = (byte)23;
-	public static final byte RBSM_MID_MEGA_STEER_FEEDBACK = (byte)24; //potentiometer 
-	public static final byte RBSM_MID_MEGA_STEER_ANGLE = (byte)20; //Arduino to servo
-	public static final byte FP_HASH = (byte)30;
-	
-	public static final byte LIGHTING_ID = (byte)50;
-	
-	public static final byte RBSM_MID_RESERVED = (byte)252;
-	public static final byte RBSM_MID_ERROR = (byte)254;
-	public static final byte RBSM_MID_DEVICE_ID = (byte)255;
```
Comment (ramensandwich) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: Did we lose the RBSM_MID header values?  They should be added into the rbsm_headers.txt along with LIGHTING_ID

Comment (ramensandwich) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: Oh nvm, I see it defined above.  I'm not 100% sure if low level will convert -1 to 255 and -2 to 254.  I'll need to test and verify that.

Comment (VivaanBahl) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: Yeah, I haven't actually tested this code fully yet, I'll need to make sure that it'll work with peel

Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: @ramensandwich can you take over testing this while vivaan is out of town?

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java (hunk 3 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128083/issues/source?bid=2938665&fileBranchId=3160458#l242)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid excessively long variable names like RBSM_HEADER_FILE_LOCATION](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128044/issues/source?bid=2938665&fileBranchId=3160458#l12)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 5]
```diff
@@ -168,30 +173,51 @@ public int peel(byte[] buffer, int start, int bytesAvailable)
...
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_COMP_HASH")) {
+			//System.out.println(message.getDataWord());
+			messagePubFp.publish(new FingerPrintMessage(message.getDataWord()));
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_MEGA_BATTERY_LEVEL")){
+			//TODO: Display the battery level in the GUI
+			//System.out.println(String.format("Battery level: %d", message.getDataWord()));
+			
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("DEVICE_ID")){
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: Codacy Issue found: [Avoid empty if statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128084/issues/source?bid=2938665&fileBranchId=3160458#l210)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 6]
```diff
@@ -168,30 +173,51 @@ public int peel(byte[] buffer, int start, int bytesAvailable)
...
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_MEGA_STEER_ANGLE")) {
+			steeringAngle = message.getDataWord();
+			messagePubControllerSteering.publish(new SteeringMeasurement(steeringAngle));
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_COMP_HASH")) {
+			//System.out.println(message.getDataWord());
+			messagePubFp.publish(new FingerPrintMessage(message.getDataWord()));
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_MEGA_BATTERY_LEVEL")){
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: Codacy Issue found: [Avoid empty if statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128084/issues/source?bid=2938665&fileBranchId=3160458#l205)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 7]
```diff
@@ -168,30 +173,51 @@ public int peel(byte[] buffer, int start, int bytesAvailable)
...
+			//System.out.println(String.format("Battery level: %d", message.getDataWord()));
+			
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("DEVICE_ID")){
+			//Stubbed out
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_MEGA_BRAKE_STATE")){
+			//Stubbed out
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_MEGA_AUTON_STATE")){
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: Codacy Issue found: [Avoid empty if statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128084/issues/source?bid=2938665&fileBranchId=3160458#l216)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 8]
```diff
@@ -168,30 +173,51 @@ public int peel(byte[] buffer, int start, int bytesAvailable)
...
+		//TODO: The following statements should be reformatted because:
+		// 1: Calling getHeaderByte so many times is inefficient
+		// 2: Header values don't change so this can be made into a switch 
+		// 3: Header values are subject to change based on what's in rbsm_headers.txt
+		if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_ENC_TICKS_RESET")) {
+			// This is a delta-distance! Do a thing!
+			encTicks = message.getDataWord() & 0xFFFF;
+			messagePubEnc.publish(estimateVelocity(message.getDataWord()));
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_ENC_TIMESTAMP")){
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: Codacy Issue found: [Avoid empty if statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128084/issues/source?bid=2938665&fileBranchId=3160458#l187)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 9]
```diff
@@ -168,30 +173,51 @@ public int peel(byte[] buffer, int start, int bytesAvailable)
...
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_MEGA_BATTERY_LEVEL")){
+			//TODO: Display the battery level in the GUI
+			//System.out.println(String.format("Battery level: %d", message.getDataWord()));
+			
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("DEVICE_ID")){
+			//Stubbed out
+		}
+		else if (headerByte == RBSerialMessage.getHeaderByte("RBSM_MID_MEGA_BRAKE_STATE")){
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: Codacy Issue found: [Avoid empty if statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128084/issues/source?bid=2938665&fileBranchId=3160458#l213)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2178128089/issues/source?bid=2938665&fileBranchId=3160458#l35)

**Comment by ramensandwich** [comment]
This has been tested and messages are being successfully received by high level.  I'll sign off on merging this.

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java (hunk 2 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java]: Codacy Issue found: [Avoid excessively long variable names like RBSM_HEADER_FILE_LOCATION](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2185905694/issues/source?bid=2938665&fileBranchId=3160458#l12)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java (hunk 1 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/serial/RBSerialMessage.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2185905739/issues/source?bid=2938665&fileBranchId=3160458#l35)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java (hunk 3 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2185905733/issues/source?bid=2938665&fileBranchId=3160458#l242)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/messages/DeviceIDMessage.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+ *          DESCRIPTION: TODO
+ *          
+ */
+public class DeviceIDMessage extends BaseMessage {
+
+	public static final String VERSION_ID = "IDV0.1";
+	private int id;
+
+	/**
+	 * Construct a new {@link BatteryLevelMessage} at time now
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/messages/DeviceIDMessage.java]: copy pasta error

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/messages/DeviceIDMessage.java [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+	 * @param idValue the id of the sending device
+	 */
+	public DeviceIDMessage(int idValue) {
+		this.id = idValue;
+		this.timestamp = new Date().getTime();
+	}
+
+	/**
+	 * Construct a new {@link BatteryMessage}
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/messages/DeviceIDMessage.java]: copy pasta

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LightingNode.java [hunk 12]
```diff
@@ -60,54 +60,11 @@ public int peel(byte[] buffer, int start, int bytesAvailable) {
...
-		vals[8] = Double.parseDouble(lightingRawStr.substring(0, hashIndex));
-		lightingRawStr = lightingRawStr.substring(hashIndex);	
-			
-//		msgPub.publish(new ImuMeasurement(vals[0], vals[1],vals[2], 
-//				vals[3], vals[4], vals[5], vals[6], vals[7], vals[8]));
-		//Feed the watchdog
-		setNodeState(NodeState.ON);
-		return 4 + (origLength - lightingRawStr.length());
+		return 1;
+
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LightingNode.java]: remove this file all together

**Pull request merged**